idx/memory/memory.go

```diff
@@ -344,21 +344,21 @@ func (m *MemoryIdx) find(orgId int, pattern string) ([]*Node, error) {
 	// for a query like foo.bar.baz, pos is 2
 	// for a query like foo.bar.* or foo.bar, pos is 1
 	// for a query like foo.b*.baz, pos is 0
-	pos := len(nodes) - 1
+	pos := len(nodes)
```

Comment:
this change makes the comments above incorrect, so comments should be updated also.